src/Inkluzitron/Services/ImagesService.cs

```diff
@@ -21,73 +25,156 @@ namespace Inkluzitron.Services
+
+        private DirectoryInfo InitializeCache(string directoryName)
         {
-            ProfilePictureService = profilePictureService;
-            Directory.CreateDirectory("ImageCache");
+            var result = new DirectoryInfo(Path.Combine(CacheSettings.DirectoryPath, directoryName));
+            result.Create();
+            return result;
         }
 
-        static public string CreateCachePath(string filename)
-            => Path.Combine("ImageCache", filename);
+        public ImagesService(DiscordSocketClient client, ImageCacheSettings cacheSettings)
+        {
+            Client = client;
+            CacheSettings = cacheSettings;
+
+            AvatarCache = InitializeCache("Avatars");
+            WhipCache = InitializeCache("Whip");
+            SpankCache = InitializeCache("Spank");
+            LoveCache = InitializeCache("Love");
+            PatCache = InitializeCache("Pat");
+            BonkCache = InitializeCache("Bonk");
+        }
 
         static private List<Bitmap> GetBitmapsFromResources<TResources>()
-        {
-            return typeof(TResources).GetProperties()
+            => typeof(TResources).GetProperties()
                 .Where(o => o.PropertyType == typeof(Bitmap))
                 .Select(o => o.GetValue(null, null) as Bitmap)
                 .Where(o => o != null)
                 .ToList();
+
+        static private string GetAvatarExtension(string avatarId)
+            => avatarId.StartsWith("a_") ? "gif" : "png";
+
+        static private AvatarImageWrapper CreateFallbackAvatarWrapper(Size? size = null)
+        {
+            var desiredSize = size ?? DefaultAvatarSize;
+
+            using var roundedFallbackAvatar = MiscellaneousResources.FallbackAvatar.RoundImage();
+            var resizedFallbackAvatar = roundedFallbackAvatar.ResizeImage(desiredSize.Width, desiredSize.Height);
+            return new AvatarImageWrapper(resizedFallbackAvatar, 1, "png");
+        }
+
+        static private readonly Size DefaultAvatarSize = new Size(100, 100);
+
+        public async Task<AvatarImageWrapper> GetAvatarAsync(SocketGuild guild, ulong userId, ushort discordSize = 128, Size? size = null)
+        {
+            if (guild == null)
+                throw new ArgumentNullException(nameof(guild));
+
+            IUser user = await guild.GetUserAsync(userId);
+            user ??= await Client.Rest.GetUserAsync(userId);
+
+            if (user == null)
+                return CreateFallbackAvatarWrapper(size);
+            else
+                return await GetAvatarAsync(user, discordSize, size);
+        }
+
+        public async Task<AvatarImageWrapper> GetAvatarAsync(IUser user, ushort discordSize = 128, Size? size = null)
+        {
+            if (user == null)
+                throw new ArgumentNullException(nameof(user));
+
+            var realSize = size ?? DefaultAvatarSize;
+            var cacheObject = FileCacheObject.In(AvatarCache)
```

Comment:
FileCacheObject je static? Nebyl byl lepší singleton v kontejneru? (Pokud je, tak ignoruj)

Reply:
`FileCacheObject.In` je jenom zkratka (statická metoda) pro `new FileCacheObjectBuilder`. Buildery jsou stavové, takže mi moc nedává smysl dávat je do DI.